### ISSUE TITLE
Fault tolerant the campaigns view

### ISSUE THREAD

**Issue by michaelbeil** [AffiliateWP/AffiliateWP#1074]
It appears that the `affiliate_wp_campaigns` view goes missing when using an export/import plugin of some variety since most of those solutions  within the WordPress ecosystem do not account for views. In one case Duplicator caused the error to come up. Also, of note is #957 where BackupBuddy is doing the same thing. So, on an import, since it’s a ‘view’ and not a table, `affiliate_wp_campaigns` might not be exported/recreated, and we should have something in place to detect that. we should properly detect that and not fail so hard or even create it if we need to. Maybe we could throw a notice stating: 

"Your campaigns are missing, to fix the issue, you can create them by installing and activating both the Debug Bar and Debug Bar Console plugins then pasting the following in Debug > Console below:

``` php   
 <?php:

$affiliate_wp_install = new stdClass();
$affiliate_wp_install->campaigns = new Affiliate_WP_Campaigns_DB;
$affiliate_wp_install->campaigns->create_view();
```
You can access this via Debug in the top right corner of the admin bar (to the left of the user profile menu): https://cloudup.com/cwfl7wgnHpf.

Otherwise, contact support for help." 

https://secure.helpscout.net/conversation/163669485/26086/?folderId=205529

**Comment by cklosowski** [comment]
If we did a notice, we'd make it have an action to create it, so the user didn't have do any other work, but perfect writeup.

**Comment by michaelbeil** [comment]
True. I also wrote that in case others need a quick reply to share with users that may have similar issues for the time being :smiley:.

**Comment by pippinsplugins** [comment]
Seems like a good idea.

**Comment by sumobi** [comment]
:+1: nice one

**Comment by michaelbeil** [comment]
:pizza:

**Comment by michaelbeil** [comment]
Also, @cklosowski deserves credit for help on this.

**Comment by DrewAPicture** [comment]
I'm kind of thinking that maybe we should add a tool to rebuild Campaigns rather than advising to run arbitrary PHP commands in the Debug Bar console.

**Comment by pippinsplugins** [comment]
I agree.

**Comment by DrewAPicture** [comment]
Bumping to 1.8.7 to allow more time for dev and testing.

**Comment by pippinsplugins** [comment]
Related #1688
